src/main/java/oc/wh40k/units/im/IMWarlordtraits.java

```diff
@@ -0,0 +1,34 @@
+package oc.wh40k.units.im;
+
+import oc.*;
+
+public class IMWarlordtraits extends Eintrag {
+    OptionsUpgradeGruppe fk1;
+    boolean spacewolves;
+
+    public IMWarlordtraits() {
+    	spacewolves = BuildaHQ.aktBuildaVater.getFormationType().equals("Space Wolves");
+        name = "Warlord Traits";
+        grundkosten = 0;
+        power = 0;
+
+        if(spacewolves) {
```

Comment:
I had used only one class of warlord traits for the complete imperial faction, like there is only one class for psy, and added all traits. That would be more maintainable than having like another 20 warlord classes for each imperial sub faction.

Regarding the dreadnought. If you know a way to utilize the used reflection style to have a different dreadnought for each faction without renaming them, we can do it by inheriting from a base class. In fact there exists no SW dreadnought or SM dreadnought by name in any army list. Adding such prefixes for every unit with a variation would lead to questions by every second user.